core/Config.php

```diff
@@ -514,7 +514,7 @@ protected function getUncached($class, $name, $sourceOptions, &$result, $suppres
 		// Include extensions only if not flagged not to, and some have been set
 		if (($sourceOptions & self::EXCLUDE_EXTRA_SOURCES) != self::EXCLUDE_EXTRA_SOURCES) {
 			// If we don't have a fresh list of extra sources, get it from the class itself
-			if (!array_key_exists($class, $this->extraConfigSources)) {
+			if (!isset($class[$this->extraConfigSources]) || $class[$this->extraConfigSources] === null) {
```

Comment:
The null check is redundant, since `isset` returns false for null values in an array.

If you want to treat "null" values as set, you need to use array_key_exists, as the code used to be, otherwise you are changing the behaviour of the code anyway, and it's not an optimisation anymore.

Before: Null values would not invoke the next line (array_key_exists is true), but after your change they would (isset is false and `!` would flag it true).

This would potentially be LESS optimal, since null values would no longer be cached, and Object:: get_extra_config_sources() would have to be run more often just to return null again.